mesonbuild/msetup.py

```diff
@@ -11,6 +12,21 @@
 from . import build, coredata, environment, interpreter, mesonlib, mintro, mlog
 from .mesonlib import MesonException
 
+if T.TYPE_CHECKING:
+    from typing_extensions import Protocol
+    from .coredata import SharedCMDOptions
+
+    class CMDOptions(SharedCMDOptions, Protocol):
+
+        profile: bool
+        fatal_warnings: bool
+        reconfigure: bool
+        wipe: bool
+        clearcache: bool
+        builddir: str
+        sourcedir: str
+        pager: bool  # this is needed for mconf compatibility
```

Comment:
Is there some other way to represent this? Like maybe a PagerMixin for mconfigure?

Reply:
So, we're sorta lying here... because there actually isn't a pager option in msetup, but the code in mconf can't be hit from msetup. I'm not sure what the better way to represent this is

Reply:
The more I think about it it seems like a heavy thing to have a mixin to share one option with an elementary type. I've updated the comments to consolidate them together in once place. Not sure though

Reply:
Sounds good. Let's see what Eli says.